### ISSUE TITLE
Why wont this run as a temporary addon in Firefox release about:debugging?

### ISSUE THREAD

**Issue by mavink** [numirias/paxmod#83]
![image](https://user-images.githubusercontent.com/5507645/123511332-440fe000-d69e-11eb-977c-138f07cf8f33.png)

![image](https://user-images.githubusercontent.com/5507645/123511346-5d189100-d69e-11eb-80e1-4a0d3cbb4e73.png)

On startup there is an uncaught type error in browser.stylesheet and again when I try to change something in the options and apply it.

This is a fresh profile with no other extensions. Nothing changes in the tab bar even on applying the options.

**Comment by numirias** [comment]
You need Firefox Developer or Nightly to run the extension because it bundles unsupported APIs ("API experiments"). Have a look at the README for the requirements.

**Comment by mavink** [comment]
You mean firefox release won't allow it even as a temporary addon? Is there any way to get it to run in firefox release? Hack omni.ja to disable addon signing?

**Comment by numirias** [comment]
> You mean firefox release won't allow it even as a temporary addon? 

I believe so. You may need to dig around in the source code to see where exactly it's getting blocked.

The error you're seeing (`browser.stylesheet` is undefined) happens because the ["stylesheet" API](https://github.com/numirias/paxmod/tree/master/src/stylesheet-api-experiment) bundled with the extension didn't get loaded. Assuming that you have set `extensions.experiment.enabled` to true already, this seems to indicate that vanilla FF won't even temporarily allow API experiments from untrusted sources.

**Comment by mavink** [comment]
Yes you are right. It's disabled from here https://searchfox.org/mozilla-central/rev/014fe72eaba26dcf6082fb9bbaf208f97a38594e/toolkit/mozapps/extensions/internal/AddonSettings.jsm#93


This works!
![image](https://user-images.githubusercontent.com/5507645/123515522-6103dd80-d6b5-11eb-9e1b-1e58fadadc6f.png)
You need only the zip file from there and paste it in the firefox install folder. Then enable the two preferences -> signing and experimental as indicated in your readme. 

There's nothing really in that zip file. Just a configuration line to disable addon signing checks. It still warns if the addon is unsigned. 

Then paxmod works in Firefox release too! Could you add this to your README so that release users can also enjoy your addon? Thanks!

**Comment by mavink** [comment]
![image](https://user-images.githubusercontent.com/5507645/123515749-6150a880-d6b6-11eb-99d3-f35ebbc741da.png)
Everything works perfectly in the release channel firefox now! Thankyou!

**Comment by numirias** [comment]
Thanks for looking into this, and great you found a solution!

I find fiddling with installation files a little awkward and potentially prone to new issues, so I'd rather want to explore some alternatives before recommending this workaround. However, I'll leave the issue open, and it's really helpful that you pointed out the general approach to bypassing the signature requirement.

**Comment by mavink** [comment]
<html><body>


extensions.experiments.enabled | true
-- | --



</body>
</html>

Needs to be set **before** installing paxmod in a fresh profile or fresh install. Otherwise it doesn't work. If you installed it before enabling the preference in about:config, you should uninstall the addon, enable the preference and then install it again. (A restart of firefox after changing the preference may be required)

**Comment by mavink** [comment]
It's a real shame that addons like this https://addons.mozilla.org/en-US/firefox/addon/native-ext/ which can make full changes to the entire system are signed and allowed, but modifying the chrome ui is not allowed! This extension allows the restyle extension [ https://addons.mozilla.org/en-US/firefox/addon/re-style/ ] to modify the tab bar etc, by modifying userchrome.css, although a restart is required everytime.

Go figure! 🙄

**Comment by mad-ady** [comment]
So - I am trying to re-enable paxmod in Firefox Release (v90) running on Ubuntu 20.04. It used to work fine until firefox updated (don't remember what the last version was).
I've done the recommendations by @mavink, but I was unsure where the zip file needed to be unpacked in Ubuntu. I tried in `/usr/lib/firefox` and `/usr/lib/firefox/browser`.

However, since I've uninstalled paxmod, I am unable to install it from file because unsigned extensions are not supported in release: https://support.mozilla.org/en-US/kb/add-on-signing-in-firefox?as=u&utm_source=inproduct. (the recommended config options in the README are set, but it seems they are not checked by the release version).

I am able to try to load it as a temporary extension, but I still get the exception mentioned in the first post: `Uncaught (in promise) TypeError: browser.stylesheet is undefined`. Most likely the zip file was not applied correctly.

So - where would I unzip that zip file on a linux system?
Thanks!

**Comment by mavink** [comment]
Please see this if it helps https://github.com/xiaoxiaoflood/firefox-scripts/issues/8#issuecomment-467619800

**Comment by mad-ady** [comment]
@mavink - thanks - I adjusted the paths and put unzipped fx-folder.zip and moved the files in the correct folders, restarted Firefox release, unchecked and reenabled the two settings (xpinstall.signatures.required = false  and extensions.experiments.enabled = true), tried to load paxmod, but it still complained that it is unsigned. If I try to load it in debug mode it complains about `browser.stylesheet is undefined`... So, it doesn't work for me.

I'd like to ask - it's unlikely to get it to install in newer FF releases, right, since it's (and will remain) unsigned? Or will it get signed once the APIs used are mature enough not to be experimental anymore?

Thanks!

**Comment by numirias** [comment]
> I'd like to ask - it's unlikely to get it to install in newer FF releases, right, since it's (and will remain) unsigned? Or will it get signed once the APIs used are mature enough not to be experimental anymore?

@mad-ady Yes, it's very unlikely that add-ons like this one will get signed by Mozilla. You'll always need some workaround to get unofficial add-ons installed.

If you're going through all the trouble to get it installed, why not just switch to FF Developer? It's basically FF Beta with a few extra features and works just fine with the README instructions.

**Comment by mad-ady** [comment]
Thank you! I think I'll have to make the switch. I can't live with a single
row of tabs, like a barbarian...

**Comment by mavinkai** [comment]
In case it does not work in FF release especially after v108. 
1. Update to the latest unsigned addon enabler - https://raw.githubusercontent.com/xiaoxiaoflood/firefox-scripts/master/fx-folder.zip (Just extract the zip into your Firefox install folder typically `C:\Program Files\Mozilla Firefox`)
2. Go to about:support and clear startup cache and restart.
3. If paxmod is still shown as disabled just install the previous release https://github.com/numirias/paxmod/releases/download/v3.15/paxmod-3.15.xpi and then install the latest release https://github.com/numirias/paxmod/releases/download/v3.16/paxmod-3.16.xpi (If FF marks an addon as unverified disabled then it has to be uninstalled and reinstalled to enable it again even after (re)enabling unsigned addons. However doing that loses all the addon configuration data. The above method of switching between versions also achieves the same thing, but without losing configuration data.)

**Comment by mavink** [comment]
Okay fixed the issue once more. The unsigned addon install feature was suddenly removed from https://github.com/xiaoxiaoflood/firefox-scripts maybe by mistake.

See https://github.com/xiaoxiaoflood/firefox-scripts/issues/324

After installing the latest firefox-scripts replace the config.js file with the one attached here (remove the .txt extension)
[config.js.txt](https://github.com/user-attachments/files/17508540/config.js.txt)

Ensure `extensions.experiment.enabled` is set to true in `about:config`

Restart firefox after replacing config.js. Clear the startupcache in about:support. Restart again.

Only after this install paxmod from file or from github. If you already installed then uninstall and reinstall. You can install tabmixplus also. Everything works perfectly. Beautiful rounded tabs with a nice black border and the newtab button from paxmod. and active/inactive tab colours from tabmixplus.

![image](https://github.com/user-attachments/assets/0be6436a-14fa-488f-9f23-fcd477f56321)

**Comment by 117649** [comment]
Use any webextension addon?

**Comment by mavink** [comment]
No APIs available to modify userchrome.css and the tab bar via webextension. Need to access privileged APIs I think

**Comment by 117649** [comment]
> No APIs available to modify userchrome.css and the tab bar via webextension. Need to access privileged APIs I think

No. 

I hope you understand why sign check can't be disabled by default now

**Comment by mavink** [comment]
It can be disabled by adding the following code as mentioned here https://github.com/xiaoxiaoflood/firefox-scripts/issues/324
